### ISSUE TITLE
Could not instiate Spec

### ISSUE THREAD

**Issue by rickle-msft** [spockframework/spock#917]
# Issue description

Running mvn test on a clean vm or in travis yields:
```
org.spockframework.util.InternalSpockError: Failed to instantiate spec 'AppendBlobAPITest'
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:369)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:275)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:239)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:160)
Caused by: java.lang.ExceptionInInitializerError
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:369)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:275)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:239)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:160)
Caused by: java.lang.NullPointerException
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:369)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:275)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:239)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:160)
```

for the first spec attempted and: 

```
org.spockframework.util.InternalSpockError: Failed to instantiate spec 'DownloadResponseTest'
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:369)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:275)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:239)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:160)
Caused by: java.lang.NoClassDefFoundError: Could not initialize class com.microsoft.azure.storage.APISpec
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:369)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:275)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:239)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:160)
```

for all subsequent specifications. I can run the tests fine in IntelliJ and on the command line on my local machine, so I'm guessing there's some configuration cached somewhere, but so far I'm unable to determine what that would be. I can point you to our pom if that would be helpful as well.

## Java/JDK
jdk1.8.0_181

## Groovy version
2.4.13

### Apache Maven
3.5.4

## Operating System
Windows

### Apache Maven
    <dependency>
      <groupId>org.spockframework</groupId>
      <artifactId>spock-core</artifactId>
      <version>1.1-groovy-2.4</version>
    </dependency>

Thank you for your help

**Comment by leonard84** [comment]
@rickle-msft you seem to be missing `com.microsoft.azure.storage.APISpec` without more information there is nothing we can do.

**Comment by rickle-msft** [comment]
Thank you for your response. The class is there, and the Groovy sources are being compiled correctly. I was able to determine that some environments weren't setting some environment variables, which was causing the NPE in initializing some fields on the type APISpec. 

Even though I was able to determine this was our error, I think it would be a good idea to improve the error messages so that if an error like this NPE is hit in the initializing the spec, the line number of the error rather than the invoke method of the JunitRunner is displayed to the user.
